beacon-chain/rpc/eth/beacon/handlers_pool.go

```diff
@@ -0,0 +1,174 @@
+package beacon
+
+import (
+	"encoding/json"
+	"net/http"
+
+	"github.com/go-playground/validator/v10"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/core/feed/operation"
+	corehelpers "github.com/prysmaticlabs/prysm/v4/beacon-chain/core/helpers"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/eth/shared"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	"github.com/prysmaticlabs/prysm/v4/crypto/bls"
+	http2 "github.com/prysmaticlabs/prysm/v4/network/http"
+	ethpbalpha "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+	"go.opencensus.io/trace"
+)
+
+// ListAttestations retrieves attestations known by the node but
+// not necessarily incorporated into any block. Allows filtering by committee index or slot.
+func (s *Server) ListAttestations(w http.ResponseWriter, r *http.Request) {
+	_, span := trace.StartSpan(r.Context(), "beacon.ListAttestations")
+	defer span.End()
+
+	rawSlot := r.URL.Query().Get("slot")
+	var slot uint64
+	if rawSlot != "" {
+		var valid bool
+		slot, valid = shared.ValidateUint(w, "Slot", rawSlot)
+		if !valid {
+			return
+		}
+	}
+	rawCommitteeIndex := r.URL.Query().Get("committee_index")
+	var committeeIndex uint64
+	if rawCommitteeIndex != "" {
+		var valid bool
+		committeeIndex, valid = shared.ValidateUint(w, "Committee index", rawCommitteeIndex)
+		if !valid {
+			return
+		}
+	}
+
+	attestations := s.AttestationsPool.AggregatedAttestations()
+	unaggAtts, err := s.AttestationsPool.UnaggregatedAttestations()
+	if err != nil {
+		http2.HandleError(w, "Could not get unaggregated attestations: "+err.Error(), http.StatusInternalServerError)
+		return
+	}
+	attestations = append(attestations, unaggAtts...)
+	isEmptyReq := rawSlot == "" && rawCommitteeIndex == ""
+	if isEmptyReq {
+		allAtts := make([]*shared.Attestation, len(attestations))
+		for i, att := range attestations {
+			allAtts[i] = shared.AttestationFromConsensus(att)
+		}
+		http2.WriteJson(w, &ListAttestationsResponse{Data: allAtts})
+		return
+	}
+
+	filteredAtts := make([]*shared.Attestation, 0, len(attestations))
+	for _, att := range attestations {
+		bothDefined := rawSlot != "" && rawCommitteeIndex != ""
+		committeeIndexMatch := rawCommitteeIndex != "" && att.Data.CommitteeIndex == primitives.CommitteeIndex(committeeIndex)
+		slotMatch := rawSlot != "" && att.Data.Slot == primitives.Slot(slot)
+
+		if bothDefined && committeeIndexMatch && slotMatch {
+			filteredAtts = append(filteredAtts, shared.AttestationFromConsensus(att))
+		} else if !bothDefined && (committeeIndexMatch || slotMatch) {
+			filteredAtts = append(filteredAtts, shared.AttestationFromConsensus(att))
+		}
```

Comment:
Since these essentially do the same thing is the reason it's split in multiple conditions for clarity? 
Rather than just being the one:
```
if (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch)) {
```
Or alternatively adding another variable:
```
toAppend := (bothDefined && committeeIndexMatch && slotMatch) || (!bothDefined && (committeeIndexMatch || slotMatch))
if toAppend {
	filteredAtts = append(filteredAtts, shared.AttestationFromConsensus(att))
} 
```

Reply:
I think it was done for more clarity, but the variable approach is also nice. I will use it.